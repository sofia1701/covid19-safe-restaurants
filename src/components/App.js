import React, { useState, useEffect } from "react";
import { Switch, Route } from "react-router-dom";
import axios from "axios";
import "../styles/App.css";
import Navbar from "./Navbar";
import Homepage from "./Homepage";
import Favourites from "./Favourites";
import Login from "./Auth/Login";
import Register from "./Auth/Register";
import Logout from "./Auth/Logout";
import Footer from "./Footer";
import userContext from "../context/userContext";

function App() {
  const [userData, setUserData] = useState({
    token: undefined,
    user: undefined,
  });

  const [query, setQuery] = useState("");

  useEffect(() => {
    const checkLoggedIn = async () => {
      let token = localStorage.getItem("auth-token");
      if (token === null) {
        localStorage.setItem("auth-token", "");
        token = "";
      }
      const tokenResponse = await axios.post(
        "https://covid-safe-auth.herokuapp.com/users/tokenIsValid",
        null,
        { headers: { "x-auth-token": token } }
      );
      if (tokenResponse.data) {
        const userResponse = await axios.get(
          "https://covid-safe-auth.herokuapp.com/users/",
          {
            headers: { "x-auth-token": token },
          }
        );
        setUserData({
          token,
          user: userResponse.data,
        });
      }
    };
    checkLoggedIn();
  }, []);

  return (
    <div className="App">
      <userContext.Provider value={{ userData, setUserData, query, setQuery }}>
        <Navbar />

        <Switch>
          <Route exact path="/" component={Homepage} />

          <Route exact path="/favourites" component={Favourites} />
          <Route exact path="/login" component={Login} />
          <Route exact path="/register" component={Register} />
          <Route exact path="/logout" component={Logout} />
        </Switch>

        <Footer />
      </userContext.Provider>
    </div>
  );
}

export default App;
