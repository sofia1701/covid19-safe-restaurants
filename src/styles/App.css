body {
  margin: 0;
  padding: 0;
}

.App {
  text-align: center;
  font-family: 'Raleway', sans-serif;
  color: black;
  display: grid;
  grid-gap: 1vh;
  min-height: 100vh;
  letter-spacing: 1px;
}

