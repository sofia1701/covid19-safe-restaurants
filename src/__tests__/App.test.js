import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import App from "../components/App";

test("renders the app and all its components correctly ", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );
  const component = asFragment();

  expect(component).toMatchSnapshot();
});
