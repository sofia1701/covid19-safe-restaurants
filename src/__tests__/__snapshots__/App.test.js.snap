// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app and all its components correctly  1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav>
      <a
        class="title"
        href="/"
      >
        Title
      </a>
      <div
        class="nav-buttons"
      >
        <a
          class="nav-button"
          href="/favourites"
        >
          Favourites
        </a>
        <a
          class="nav-button"
          href="/login"
        >
          Sign In
        </a>
        <a
          class="nav-button"
          href="/register"
        >
          Register
        </a>
      </div>
    </nav>
    <div
      class="homepage"
    >
      <div
        class="search-wrapper"
      >
        <div
          class="search-info"
        >
          <h2>
            Search for a safe restaurant in Manchester
          </h2>
          <form
            class="search-box"
          >
            <button
              class="search-button"
              type="submit"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                />
              </svg>
            </button>
            <input
              class="search-input"
              placeholder="The Bay Horse Tavern..."
              type="text"
              value=""
            />
          </form>
          <h3>
            Or select your favourite cuisine
          </h3>
          <div
            class="row"
          >
            <a
              href="/"
            >
              All Restaurants
            </a>
            <a
              href="/?query={\\"type\\": \\"African\\"}"
            >
              African
            </a>
            <a
              href="/?query={\\"type\\": \\"Asian\\"}"
            >
              Asian
            </a>
            <a
              href="/?query={\\"type\\": \\"American\\"}"
            >
              American
            </a>
            <a
              href="/?query={\\"type\\": \\"British\\"}"
            >
              British
            </a>
            <a
              href="/?query={\\"type\\": \\"Mexican\\"}"
            >
              Mexican
            </a>
            <a
              href="/?query={\\"type\\": \\"Bakery\\"}"
            >
              Bakery
            </a>
          </div>
          <div
            class="row"
          >
            <a
              href="/?query={\\"type\\": \\"Brazilian\\"}"
            >
              Brazilian
            </a>
            <a
              href="/?query={\\"type\\": \\"Italian\\"}"
            >
              Italian
            </a>
            <a
              href="/?query={\\"type\\": \\"Spanish\\"}"
            >
              Spanish
            </a>
            <a
              href="/?query={\\"type\\": \\"French\\"}"
            >
              French
            </a>
            <a
              href="/?query={\\"type\\": \\"Irish\\"}"
            >
              Irish
            </a>
            <a
              href="/?query={\\"type\\": \\"Vegan\\"}"
            >
              Vegan
            </a>
            <a
              href="/?query={\\"type\\": \\"International\\"}"
            >
              International
            </a>
            <a
              href="/?query={\\"type\\": \\"Cafe\\"}"
            >
              Cafe
            </a>
          </div>
        </div>
      </div>
      <div
        class="search-results"
      >
        <div
          class="restaurants"
        />
      </div>
    </div>
    <div
      class="footer"
    >
      Footer
    </div>
  </div>
</DocumentFragment>
`;
